text/0000-rustdoc-types-maintainers.md

```diff
@@ -0,0 +1,134 @@
+
+Republishing `rustdoc-json-types` as `rustdoc-types` is done with [a script](https://github.com/aDotInTheVoid/rustdoc-types/blob/17cbe9f8f07de954261dbb9536c394381770de7b/update.sh) so that it is as low maintenance as possible. This also ensures that all format/documentation changes happen in the rust-lang/rust repo, and go through the normal review process there.
+
+The update/publishing process will be moved to T-rustdoc. In the medium term, I (`@aDotInTheVoid`) will still do it, but
+- In an official capacity
+- With bus factor for when I stop.
+
+We (T-rustdoc) will continue to publish a new version of the `rustdoc-types` crate
+every time the upstream implementation changes, and these will be versioned with
+normal SemVer. Changes to rustdoc-json in `rust-lang/rust` will not be accepted
+if they would make it not possible to publish `rustdoc-types`.
+
+## Actual Mechanics of the move
+
+### GitHub
+
+GitHub has a [list of requirements](https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository) for transferring repositories. T-infra will handle the permissions of moving the repository into the rust-lang GitHub organization.
+
+At the end of this we should have a moved the [`aDotInTheVoid/rustdoc-types`]
+repo into the rust-lang GitHub org. T-rustdoc will have `maintain` permissions
+(via the [team repo](https://github.com/rust-lang/team/)).
+
+### crates.io
+
+crates.io ownership is managed [via the command line](https://doc.rust-lang.org/cargo/reference/publishing.html#cargo-owner).
+
+I will run the following commands to move ownership.
+
+```
+cargo owner --add github:rust-lang:rustdoc
+cargo owner --add rust-lang-owner
+cargo owner --remove aDotInTheVoid
+```
+
+The `rust-lang-owner` is needed because team owners cannot add new owners. 
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Adds additional maintenance burden to rustdoc team.
+- One-time maintenance burden to infra team to support move.
+
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+- We could keep `rustdoc-types` as a personal project. This preserves the status quo (and is what will happen if this RFC (or something similar) isn't adopted). This is undesirable because
+      - Bus factor: If I am unable or unwilling to maintain `rustdoc-types`, we cause a load of unnecessary churn when it becomes out of sync with the in-tree `rustdoc-json-types`
+- We could bundle `rustdoc-types` through rustup. This is undesirable as it means users can't depend on it in stable rust, and can't depend on multiple versions.
+- We could publish `rustdoc-json-types` directly from `rust-lang/rust`. However
+   - `rust-lang/rust` doesn't currently publish to crates.io.
+   - `rustdoc-json-types` doesn't currently bump the version field in `Cargo.toml`
+   - It may be desirable to one day use different types for rustdoc serialization vs users deserialization
```

Comment:
FWIW we could use `cfg`s for this (like `feature = "unstable_internal_rustdoc"` for the internal rustdoc version). I don't think we'd want to maintain two completely separate versions of the API anyway, since it would be confusing to keep in sync.

Reply:
Yeah, we should probably do that either way, to replace the [increasing pile of regex](https://github.com/aDotInTheVoid/rustdoc-types/blob/aea8d42347f0a94c7702329147c32098181221bd/update.sh#L12-L14) we currently use.